src/web/mjs/connectors/kuaikanmanhua.mjs

```diff
@@ -20,82 +20,70 @@ export default class Kuaikanmanhua extends Connector {
+            mangaList.push(...mangas);
         }
-        return this.wait( 0 )
-            .then ( () => this.fetchDOM( mangaPageLinks[ index ], 'div.tagContent div a', 5 ) )
-            .then( data => {
-                let mangaList = data.map( element => {
-                    this.cfMailDecrypt( element );
-                    return {
-                        id: this.getRelativeLink( element ),
-                        title: element.text.trim()
-                    };
-                } );
-                if( index < mangaPageLinks.length - 1 ) {
-                    return this._getMangaListFromPages( mangaPageLinks, index + 1 )
-                        .then( mangas => mangas.concat( mangaList ) );
-                } else {
-                    return Promise.resolve( mangaList );
-                }
-            } );
+        return mangaList;
     }
 
-    _getMangaList( callback ) {
-        this.fetchDOM( this.url + this.list, 'ul.pagination li:nth-last-child(2) a' )
-            .then( data => {
-                let pageCount = parseInt( data[0].text.match( /\d+/ )[0] );
-                let pageLinks = [... new Array( pageCount ).keys()].map( page => this.url + this.list + '?page=' + ( page + 1 ) );
-                return this._getMangaListFromPages( pageLinks );
-            })
-            .then( data => {
-                callback( null, data );
-            })
-            .catch( error => {
-                console.error( error, this );
-                callback( error, undefined );
-            });
+    async _getMangasFromPage(page) {
+        const uri = new URL(this.list + '?page='+ page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.tagContent div a', 5);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
     }
 
-    async _getChapterList(manga, callback) {
-        try {
-            let request = new Request(this.url + manga.id, this.requestOptions);
-            let data = await this.fetchDOM(request, 'div.TopicList div div.TopicItem div.title a');
-            let chapterList = data.map(element => {
-                return {
-                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
-                    title: element.text.trim(),
-                    language: ''
-                };
-            }).filter(chapter => chapter.id !== 'javascript:void(0);');
-            callback(null, chapterList);
-        } catch (error) {
-            console.error(error, manga);
-            callback(error, undefined);
+    async _getChapters(manga) {
+        let chapterList = await this.resolveChapterList(manga, 'comics');
+        if (chapterList.length < 1) {
+            chapterList = await this.resolveChapterList(manga, 'comicList');
```

Comment:
I feel that those double calls should be done within a single call to resolveChapterList (in the fetchUI script . And then you can just write the call here)